keras_cv/models/feature_extractor/clip/clip_image_model.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from keras_cv.backend import keras
+from keras_cv.backend import ops
+from keras_cv.models.feature_extractor.clip.clip_encoder import CLIPEncoder
+from keras_cv.models.feature_extractor.clip.clip_encoder import get_initializer
+
+
+class CLIPPatchingAndEmbedding(keras.layers.Layer):
+    def __init__(
+        self, width, patch_size, input_resolution, output_dim, **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.conv1 = keras.layers.Conv2D(
+            filters=width,
+            kernel_size=patch_size,
+            strides=patch_size,
+            padding="valid",
+            use_bias=False,
+            data_format="channels_last",
+            kernel_initializer=get_initializer(0.02),
+            name="patch_embed.embedding",
+        )
+        self.width = width
+        self.input_resolution = input_resolution
+        self.patch_size = patch_size
+        self.num_patches = ops.power(
+            (self.input_resolution // self.patch_size), 2
+        )
+        self.class_embedding_initializer = get_initializer(
+            ops.power(self.width, -0.5) * 0.02
+        )
+        self.output_dim = output_dim
+
+    def build(self, input_shape):
+        self.conv1.build(input_shape)
```

Comment:
We don't need `super().build(..)` here?  I see them in other CLIP Layers

Reply:
added!